### DISCUSSION TITLE
Is it able to install dependencies through local .whl file instead of fetching it from the Internet and build it locally?

### DISCUSSION THREAD

**Discussion by saddestboy** [astral-sh/rye#558] [General]
Well, some dependencies are hard to build on the computer (especially windows) since it requires MSVC and specific C include files, so it would be helpful to use local .whl file instead of building it on the computer. However I cannot find such description in the docs, so is it able to install package through local .whl file instead of fetching it from the Internet and build it locally?

**Comment by mitsuhiko** [comment]
Yes it's possible. You should be able to register a `file:///` path with a `find-links` type index.

**Comment by NathanCummings** [comment]
I also have this use-case. I'm afraid, despite your answer, I still don't know how I'd go about doing this. I looked at the docs (https://rye-up.com/guide/sources/#index-authentication) but I still don't know how to do it.

**Reply by mitsuhiko** [reply]
You add a new index to your `pyproject.toml` like this:

```toml
[[sources]]
name = "a-name"
url = "file:///path/to/your/stuff"
```

and then that folder is considered for resolution

**Reply by NathanCummings** [reply]
Much appreciated, I'll give it a try 😃 

Also, thank you for your work on this. Only got started with it yesterday but I'm already really impressed!

**Comment by NathanCummings** [comment]
Ah, I don't think it will work. It's not a normal python project, all I have is a .whl, so I get this error:
<img width="1095" alt="image" src="https://github.com/mitsuhiko/rye/assets/29842194/138e7c17-4909-4727-81e1-6cb5ff6e81b4">
